src/Adyen/Service.php

```diff
@@ -43,11 +43,51 @@ public function getClient()
         return $this->client;
     }
 
+    /**
+     * @throws AdyenException
+     */
+    protected function requestHttp($url, $method = 'get', array $bodyParams = null, $requestOptions = null)
+    {
+        // check if rest api method has a value
+        if (!$method) {
+            $msg = 'The REST API method is empty';
+            $this->getClient()->getLogger()->error($msg);
+            throw new AdyenException($msg);
+        }
+        // check if rest api method has a value
+        if (!$url) {
+            $msg = 'The REST API endpoint is empty';
+            $this->getClient()->getLogger()->error($msg);
+            throw new AdyenException($msg);
+        }
+        $curlClient = $this->getClient()->getHttpClient();
+        return $curlClient->requestHttpRest($this, $url, $bodyParams, $method, $requestOptions);
+    }
+
     /**
      * @return bool
      */
     public function requiresApiKey()
     {
         return $this->requiresApiKey;
     }
+
+    /**
+     * @param string $url
+     * @return string
+     */
+    protected function createBaseUrl(string $url): string
+    {
+        $config = $this->getClient()->getConfig();
+        if ($this->getClient()->getConfig()->get('enableLive')) {
```

Comment:
I would rather to avoid adding this new `enableLive` config. 

```suggestion
        if ($this->getClient()->getConfig()->getEnvironment() === \Adyen\Environment::LIVE) {
```